WIP add guidance for notification banner

edwardhorsford opened the pull request:
WIP

Still needs examples, adapted content

sarawilcox commented:
Thanks for this, @edwardhorsford. It's looking good. :-)

I've made some changes but have some queries too. (I mentioned some of them on Slack.)

The GOV page talks about messages only in emergency. They use a COVID example but you mentioned it may be more about systems outages. I've avoided mentioning emergencies as there could be other reasons.(Though, as you said, the confirmation page might be a better place for this sort of messaging. I can put the wording around emergencies back in if you think we should.

I’ve had a go at explaining that the component includes an H2 (usually) and why GOV says to use H3 for further headings. Does what I've said work?

GOV's component links to Service unavailable and There is problem with this service patterns, which we don’t have. Should we link to GOV or not mention them?

Is the "Patient outcome recorded" example a tested one? (It doesn’t have to be but helps if it is.)

I wondered if the green success example should sit under the heading “Reacting to something the user has done” after the 1st sentence, followed by examples of when not to use it, e.g. in linear journeys.

Re the WCAG2.2 info at the bottom of the page, we normally use the WCAG 2.2 blue tags (see [buttons](https://service-manual.nhs.uk/design-system/components/buttons) page). GOV got rid of them. Do we want to use our standard WCAG 2.2 treatment? 

I've not done the change log or What’s new pages yet. But I have added this page to the site maps.

edwardhorsford commented on app/views/design-system/components/notification-banner/index.njk:
```diff
@@ -0,0 +1,126 @@
+
+<p>If you need another heading, place it in <code>nhsuk-notification-banner__content</code>. It will normally be an <code>&lt;h3&gt;</code>.</p>
+
+<p>Avoid using headings for single-line notifications that do not need them.</p>
```
Could note something that the styles support stuff that _visually_ looks like a heading, it's just that the markup doesn't need to be one?

anandamaryon1 replied:
> When not using a heading, you can style the text to stand out, for example, by applying the `nhsuk-notification-banner__heading` class to a paragraph `<p>`. 

Maybe this?

edwardhorsford commented on app/views/design-system/components/notification-banner/index.njk:
```diff
@@ -0,0 +1,126 @@
+
+<h2 id="telling-the-user-about-a-problem-that-affects-the-whole-service">Telling the user about a problem that affects the whole service</h2>
+
+<p>Use a "neutral" blue notification banner if the user needs to know about a problem with the service as a whole.</p>
```
Might mention that 'information' can be used for other cases than problems. Eg system update messages.

edwardhorsford commented:
> Thanks for this, @edwardhorsford. It's looking good. :-)
> 
> I've made some changes but have some queries too. (I mentioned some of them on Slack.)
> 
> The GOV page talks about messages only in emergency. They use a COVID example but you mentioned it may be more about systems outages. I've avoided mentioning emergencies as there could be other reasons.(Though, as you said, the confirmation page might be a better place for this sort of messaging. I can put the wording around emergencies back in if you think we should.
> 
> I’ve had a go at explaining that the component includes an H2 (usually) and why GOV says to use H3 for further headings. Does what I've said work?
>
Added some comments inline
> GOV's component links to Service unavailable and There is problem with this service patterns, which we don’t have. Should we link to GOV or not mention them?

Do we have precedent of linking to govuk things where we don't have a page? If so I'd keep them, but otherwise we could remove?
> 
> Is the "Patient outcome recorded" example a tested one? (It doesn’t have to be but helps if it is.)

No, I made it up. It's a bit similar to this that we have in my service (still in development).

<img width="1734" height="444" alt="Screenshot of green success notification banner" src="https://github.com/user-attachments/assets/c919bfd7-c7f6-4e4e-b3cb-83cb9e30d043" />

> I wondered if the green success example should sit under the heading “Reacting to something the user has done” after the 1st sentence, followed by examples of when not to use it, e.g. in linear journeys.

Sounds good!

> Re the WCAG2.2 info at the bottom of the page, we normally use the WCAG 2.2 blue tags (see [buttons](https://service-manual.nhs.uk/design-system/components/buttons) page). GOV got rid of them. Do we want to use our standard WCAG 2.2 treatment?

Let's change for whatever our current format is.

> I've not done the change log or What’s new pages yet. But I have added this page to the site maps.

Thanks! I think I meant to search where else this needs linking but forgot.

anandamaryon1 commented on app/views/design-system/components/notification-banner/index.njk:
```diff
@@ -0,0 +1,126 @@
+
+<p>You can also use a notification banner to tell the user about the outcome of something they've just done. If they have not finished the current journey, it does not make sense to use a <a href="/design-system/patterns/confirmation-page/">confirmation page</a> at this point in your service.</p>
+
+<p>Using a notification banner is unlikely to be the right approach in a linear service, for example, a service that lets the user register for a thing. For a linear service, it usually makes sense to stick to the approach of <a href="https://www.gov.uk/service-manual/design/form-structure#start-with-one-thing-per-page">starting with 1 thing per page (in GOV.UK's guidance on structuring forms)</a ...
```
Could link in usage of the task list, for multi step journeys. And to mention that a success banner shouldn't be used on a task list page. (or should it? I'd assume no, since the status is communicated by the tags)

edwardhorsford commented on app/views/design-system/components/notification-banner/index.njk:
```diff
@@ -22,47 +22,57 @@
+<li>a green success banner</li>
 </ul>
 
+<p>Use the blue banner to tell the user information about:</p>
```
I'd refer to them by the semantic name (information banner) rather than the colour.

edwardhorsford commented on app/views/design-system/components/notification-banner/index.njk:
```diff
@@ -22,47 +22,57 @@
+<li>something that affects them in particular (for example, an approaching deadline they need to meet)</li>
+</ul>
+
+<p>Use the green banner to tell the user that something they've just done on a previous page was successful (for example, confirming that an email has been sent).</p>
```
As above, would refer to this as the success banner, not the green banner.

sarawilcox replied:
I'll update the branch to call them success banner and info banner.

edwardhorsford commented on app/views/design-system/components/notification-banner/index.njk:
```diff
@@ -22,47 +22,57 @@
-<p>Use notification banners sparingly. There's evidence that people often miss banners, and using them too often is likely to make this problem worse. See <a href="https://www.nngroup.com/articles/banner-blindness-old-and-new-findings/">Banner blindness revisited by Norman Nielsen Group</a>.</p>
+<p>Use notification banners sparingly. There's evidence that people often miss banners. Using them too often is likely to make this problem worse. See <a href="https://www.nngroup.com/articles/banner-blindness-old-and-new-findings/">Banner blindness revisited by Norman Nielsen Group</a>.</p>
+
+<p>Avoid showing more than 1 notification banner on the same page. Instead, combine the messages in a single notification banner. If the messages are too different to combine, only show the highest priority notification banner.</p>
```
> If the messages are too different to combine, only show the highest priority notification banner.

I suspect this could be rather hard to actually make work in practice due to how banners are usually implemented.

sarawilcox replied:
I take this to mean, only have 1 banner with your highest priority message.

edwardhorsford commented on app/views/design-system/components/notification-banner/index.njk:
```diff
@@ -22,47 +22,57 @@
 <h2 id="how-it-works">How it works</h2>
 
-<p>Position a notification banner immediately before the page <code>h1</code>. The notification banner should be the same width as the page's other content, such as components, headings and body text. For example, if the other content takes up two-thirds of the screen on desktop devices, then the notification banner should also take up two-thirds. <a href="/design-system/styles/layout/">Read abou ...
+<p>Put a notification banner immediately before the page <code>h1</code>. The notification banner should be the same width as the page's other content, such as components, headings and body text. For example, if the other content takes up two-thirds of the screen on desktop devices, then the notification banner should also take up two-thirds. <a href="/design-system/styles/layout/">Read about pag ...
```
I think the banner wants to get placed in the same position as the error summary. The error summary suggests this for placement:

>Put the error summary at the top of the main container. If your page includes breadcrumbs or a back link, place it below these, but above the <h1>.

edwardhorsford commented on app/views/design-system/components/notification-banner/index.njk:
```diff
@@ -22,47 +22,57 @@
+<p>When you're not using a heading, you can style the text to stand out, for example, by applying the <code>nhsuk-notification-banner__heading</code> class to a paragraph.</p>
 
-<h2 id="telling-the-user-about-a-problem-that-affects-the-whole-service">Telling the user about a problem that affects the whole service</h2>
+<h2 id="using-the-blue-information-banner">Using the blue notification banner</h2>
```
As above, would refer to this as the information banner.

sarawilcox commented:
I'm not sure about this wording re the notification banner as a whole.

"A notification banner lets you tell the user about something that's not directly relevant to the thing they're trying to do on that page of the service." The success banner is related to something you've just done.

anandamaryon1 commented:
> I'm not sure about this wording re the notification banner as a whole.
> 
> "A notification banner lets you tell the user about something that's not directly relevant to the thing they're trying to do on that page of the service." The success banner is related to something you've just done.

Agree, I think it's more about the content of the page their on, rather than relation to their task. So a success banner would be on the next page when you, eg, create a new thing. Not on the top of the page where you made the thing. That's my understanding anyway.

colinrotherham commented:
I've rebased this PR branch on top of our v10 breaking changes PR:

* https://github.com/nhsuk/nhsuk-frontend/pull/1408

i.e. It'll be combined with https://github.com/nhsuk/nhsuk-frontend/pull/1408 once merged
